src/game/client/water/w3dwater.h

```diff
@@ -0,0 +1,225 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Water Render Object
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include "asciistring.h"
+#include "gametype.h"
+#include "rendobj.h"
+#include "shader.h"
+#include "snapshot.h"
+#include "vector3.h"
+
+class DX8IndexBufferClass;
+class SceneClass;
+class VertexMaterialClass;
+class LightClass;
+class TextureClass;
+class WaterTracksRenderSystem;
+class PolygonTrigger;
+
+struct IDirect3DDevice8;
+struct IDirect3DVertexBuffer8;
+struct IDirect3DIndexBuffer8;
+struct IDirect3DTexture8;
+
+struct VertexFormatXYZDUV2
+{
+    float x;
+    float y;
+    float z;
+    unsigned int diffuse;
+    float u1;
+    float v1;
+    float u2;
+    float v2;
+};
+
+class WaterRenderObjClass : public SnapShot, public RenderObjClass
+{
+    ALLOW_HOOKING
+
+public:
+    enum WaterType
+    {
+        WATER_TYPE_0_TRANSLUCENT = 0,
+        WATER_TYPE_1_FB_REFLECTION,
+        WATER_TYPE_2_PVSHADER,
+        WATER_TYPE_3_GRIDMESH,
+        WATER_TYPE_MAX,
+    };
+
+    struct WaterMeshData
+    {
+        float height;
+        float velocity;
+        unsigned char status;
+        unsigned char preferred_height;
+    };
+
+    struct Setting
+    {
+        TextureClass *sky_texture;
+        TextureClass *water_texture;
+        int water_repeat_count;
+        float sky_texels_per_unit;
+        unsigned long vertex00_diffuse;
+        unsigned long vertex10_diffuse;
+        unsigned long vertex11_diffuse;
+        unsigned long vertex01_diffuse;
+        unsigned long water_diffuse;
+        unsigned long transparent_water_diffuse;
+        float u_scroll_per_ms;
+        float v_scroll_per_ms;
+    };
+
+    struct SEA_PATCH_VERTEX

```

Comment:
Do we really need 6+ identical structs? Just do a: `using SeaPatchVertex = VertexFormatXYZDUV1` or whatever the other struct is called

Reply:
ahh wait this is a Vector3.